imap_data_access/processing_input.py

```diff
@@ -370,17 +376,30 @@ def deserialize(self, json_input: str) -> None:
             elif file_creator["type"] == ProcessingInputType.SPICE_FILE.value:
                 self.add(SPICEInput(*file_creator["files"]))
 
-    def get_science_files(self) -> list[ProcessingInput]:
-        """Return just the science files from the collection.
+    def get_files(self, instrument, descriptor) -> list[Path]:
+        """Get the dependency files path from the collection.
 
         Returns
         -------
-        out : list[ProcessingInput]
+        out : list[Path]
             list of ScienceInput files contained in the collection.
         """
         out = []
         for file in self.processing_input:
-            if file.input_type == ProcessingInputType.SCIENCE_FILE:
-                out.append(file)
-
+            if file.source == instrument and file.descriptor == descriptor:
+                for obj in file.imap_file_paths:
+                    out.append(obj.construct_path())
         return out
+
+    def donwload(self):
+        """Download all the dependencies for the processing input."""
+        # Go through science or ancillary or SPICE dependencies
+        # processing input list and download all files
+        for dependency in self.processing_input:
+            for filepath in dependency.imap_file_paths:
+                try:
+                    download_path = filepath.construct_path()
+                    logger.info(f"Downloading {download_path}")
```

Comment:
Do we need to log here or can we rely on the `download()` function itself for the logging? Maybe we could add more logs there if necessary, but I think there are some info logs already in `download()`.